### DISCUSSION TITLE
Applying Reinforcement Learning technique to RPL protocol

### DISCUSSION THREAD

**Discussion by NajimDH** [contiki-ng/contiki-ng#2445] [Ideas]
Hello, 
Based on many papers, for **routing functionality** in IoT-WSN networks, we can utilize artificial intelligence algorithms to provide the routing protocol with self-learning and self-adaptive capabilities. Can anyone please give guidance about how to apply Reinforcement Learning to **RPL protocol** using the **COOJA** simulator?
thanks in advance.

**Comment by khanhongithub** [comment]
I am also looking for an example

**Comment by NajimDH** [comment]
Hi, Because there is a little feedback, if you like we can work together on this problem.

**Comment by khanhongithub** [comment]
Hey for sure. Im looking forward to discussing it

**Comment by khanhongithub** [comment]
Hi NajimDH,

I am Khanh. I am looking forward to discussing this with you.
We can connect on WhatsApp as well. This is my Whatsapp number:
+4915202104744

Cheers,
Khanh

**Comment by NajimDH** [comment]
Hi Khanh,

Thank you for your reply, I will send you via WhatsApp.

Regards,

**Reply by Djihene29Butterfly** [reply]
Hi Najim,
Did you find any hints may help to implement reinforcement learning in cooja simulator?

**Comment by vishnuvardhan369** [comment]
Hey can you guys help me out in this topic. I am also doing a project on this topic.

**Comment by dengcen** [comment]
I am  looking forward to discussing it, too.  The my email:dcen1980@163.com. I wish we can discuss together for this topic.
